### PR TITLE
Update Armadillo to minimum version of 8.400.0.

### PR DESCRIPTION
**Pull request by rcurtin**
This should fix the Travis build, updates the documentation, and should also remove some now-unnecessary code.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ matrix:
         - sudo apt-get update
         - sudo apt-get install -y --allow-unauthenticated libopenblas-dev liblapack-dev g++ libboost-all-dev python-pip cython python-numpy python-pandas
         - sudo pip install --upgrade --ignore-installed setuptools cython
-        - curl https://ftp.fau.de/macports/distfiles/armadillo/armadillo-6.500.5.tar.gz | tar xvz && cd armadillo*
+        - curl https://ftp.fau.de/macports/distfiles/armadillo/armadillo-8.400.0.tar.xz | tar xv && cd armadillo*
```
Comment (zoq) [inline .travis.yml]: In ensmallen I used:

``curl https://ftp.fau.de/macports/distfiles/armadillo/armadillo-8.400.0.tar.xz | tar -xvJ && cd armadillo*;``

and also installed ``xz-utils`` not sure that step is necessary.

Comment (rcurtin) [reply .travis.yml]: Yeah, I made the same change here... let's see if it works.  I don't mind installing `xz-utils` even if it's not needed. :)

**Comment by KimSangYeon-DGU** [comment]
Looks good to me.

**Comment by zoq** [comment]
Nice, let's restart some travis builds.

**Pull request merged**